Tests/QueryTests.swift

```diff
@@ -0,0 +1,343 @@
+                expect(cats.first!.likes).to(contain("fish"))
+
+            case .error:
+                fail("Should not throw an error")
+            }
+            expectation.fulfill()
+        }
+
+        waitForExpectations(timeout: 10.0, handler: nil)
+    }
+
+    func testExistenceQuery() {
+        let expectation = self.expectation(description: "Existence operator expectation")
+
+        let query = Query<Cat>.query(where: "fields.color", .exists(true))
+
+        QueryTests.client.fetchEntries(with: query) { result in
+            switch result {
+            case .success(let cats):
+                expect(cats.count).to(beGreaterThan(0))
+                expect(cats.first!.color).toNot(equal("gray"))
+            case .error:
+                fail("Should not throw an error")
+            }
+            expectation.fulfill()
+        }
+
+        waitForExpectations(timeout: 10.0, handler: nil)
+    }
+
+    func testChainingQueries() {
+
+        let expectation = self.expectation(description: "Chained operator expectation")
+
+        let query = Query<Cat>.query(where: "fields.color", .doesNotEqual("gray"))
+            .query(where: "fields.lives", .equals("9"))
+
+        QueryTests.client.fetchEntries(with: query) { result in
+            switch result {
+            case .success(let cats):
+                expect(cats.count).to(equal(1))
+                expect(cats.first!.lives).to(equal(9))
+
+            case .error(let error):
+                fail("Should not throw an error \(error)")
+            }
+            expectation.fulfill()
+        }
+
+        waitForExpectations(timeout: 10.0, handler: nil)
+    }
+
+    func testQueryAssets() {
+        let expectation = self.expectation(description: "Inequalitys operator expectation")
+
+        let query = try! Query<ImageAsset>.query(where: "sys.id", .equals("1x0xpXu4pSGS4OukSyWGUK"))
+            .select(fieldNames: ["fields.title"])
+        QueryTests.client.fetchAssets(with: query) { result in
+            switch result {
+            case .success(let imageAssets):
+                expect(imageAssets.count).to(equal(1))
+                expect(imageAssets.first!.id).to(equal("1x0xpXu4pSGS4OukSyWGUK"))
+                expect(imageAssets.first!.title).to(equal("Doge"))
+            case .error(let error):
+                fail("Should not throw an error \(error)")
+            }
+            expectation.fulfill()
+        }
+        waitForExpectations(timeout: 10.0, handler: nil)
+    }
+
+    func testQueryValidation() {
+        let fieldNames = ["sys.contentType.sys.id"]
+
+        do {
+            let _ = try Query<Dog>.select(fieldNames: fieldNames)
+
+            fail("Query selection with depth > 2 should throw an error and not reahc here")
+        } catch let error as QueryError {
+            expect(error.message).toNot(beNil())
+        } catch _ {
+            fail("Should throw a QueryError")
```

Comment:
I'm catching here to make sure that the error thrown is the right type of `Error`. So yes, I do indeed need to catch it now.